prog/minio/setup_minio.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+class Prog::Minio::SetupMinio < Prog::Base
+  subject_is :minio_server
+
+  label def install_minio
+    case minio_server.vm.sshable.cmd("common/bin/daemonizer --check install_minio")
+    when "Succeeded"
+      pop "minio is installed"
+    when "Failed", "NotStarted"
+      minio_server.vm.sshable.cmd("common/bin/daemonizer 'minio/bin/install_minio' install_minio")
+    end
+    nap 5
+  end
+
+  label def configure_minio
+    minio_server.vm.sshable.cmd(<<SH)
+set -euo pipefail
+sudo sh -c 'echo "MINIO_VOLUMES="#{minio_server.minio_volumes}"" > /etc/default/minio'
+echo 'MINIO_OPTS="--console-address :9001"' | sudo tee -a /etc/default/minio
+sudo sh -c 'echo "MINIO_ROOT_USER="#{minio_server.minio_cluster.admin_user}"" >> /etc/default/minio'
+sudo sh -c 'echo "MINIO_ROOT_PASSWORD="#{minio_server.minio_cluster.admin_password}"" >> /etc/default/minio'
+echo "#{minio_server.minio_cluster.generate_etc_hosts_entry}" | sudo tee -a /etc/hosts
+sudo chown -R minio-user:minio-user /etc/default/minio
+# YYY: Will be updated once the dns is setup
+# MINIO_SERVER_URL="https://#{minio_server.minio_cluster.name}.#{Config.minio_host_name}:9000"
+SH
```

Comment:
I think it's weird that you insert a bunch of quotes instead of using shellescape somehow. Seems like you could combine the steps in setting up `/etc/default/minio` too and simplify, I'd think. You might also gain a small security advantage by avoiding `sudo` (which logs what it is given) and using `cmd(..., stdin:)` for this job. It can also make the quoting, or lack thereof, come out more elegantly.